extra/modules/optable-targeting/src/main/java/org/prebid/server/hooks/modules/optable/targeting/v1/core/BidRequestEnricher.java

```diff
@@ -60,18 +66,89 @@ private BidRequest enrichBidRequest(BidRequest bidRequest) {
+            return source;
+        }
+
+        if (CollectionUtils.isEmpty(source)) {
+            return destination;
+        }
+
+        final Map<String, Eid> idToSourceEid = source.stream().collect(Collectors.toMap(
+                BidRequestEnricher::eidIdExtractor,
+                Function.identity(),
+                (a, b) -> b,
+                HashMap::new));
+
+        final Set<String> sourceToReplace = targetingProperties.getOptableInserterEidsReplace();
+        final Set<String> sourceToMerge = targetingProperties.getOptableInserterEidsMerge()
+                .stream()
+                .filter(it -> !sourceToReplace.contains(it)).collect(Collectors.toSet());
+
+        final List<Eid> mergedEid = destination.stream()
+                .map(destinationEid -> idToSourceEid.containsKey(eidIdExtractor(destinationEid))
+                        && OPTABLE_CO_INSERTER.equals(destinationEid.getInserter())
+                        ? resolveEidConflict(
+                                destinationEid,
+                                idToSourceEid.get(eidIdExtractor(destinationEid)),
+                                sourceToMerge,
+                                sourceToReplace)
+                        : destinationEid)
+                .toList();
+
+        return merge(mergedEid, source, BidRequestEnricher::eidIdExtractor);
+    }
+
+    private List<Eid> filterOptableEids(List<Eid> eids) {
+        if (CollectionUtils.isEmpty(eids)) {
+            return eids;
+        }
+
+        final Set<String> optableIdsToIgnore = targetingProperties.getOptableInserterEidsIgnore();
+        if (CollectionUtils.isEmpty(optableIdsToIgnore)) {
+            return eids;
+        }
+
+        return eids.stream()
+                .filter(eid -> !OPTABLE_CO_INSERTER.equals(eid.getInserter())
+                        || !optableIdsToIgnore.contains(eid.getSource()))
+                .toList();
+    }
+
+    private static Eid resolveEidConflict(Eid destinationEid,
+                                          Eid sourceEid,
+                                          Set<String> sourceToMerge,
+                                          Set<String> sourceToReplace) {
+
+        final String eidSource = sourceEid.getSource();
+
+        if (sourceToReplace.contains(eidSource)) {
+            return sourceEid;
+        }
+        if (sourceToMerge.contains(eidSource)) {
+            return mergeEid(destinationEid, sourceEid);
+        }
+
+        return destinationEid;
+    }
+
+    private static Eid mergeEid(Eid destinationEid, Eid sourceEid) {
+        return destinationEid.toBuilder()
+                .uids(merge(destinationEid.getUids(), sourceEid.getUids(), Uid::getId))
+                .build();
+    }
+
+    private static String eidIdExtractor(Eid eid) {
+        final String inserter = Optional.ofNullable(eid.getInserter()).orElse("");
+        return "%s_%s".formatted(inserter, eid.getSource());
```

Comment:
Previous approach with `ObjectUtil.getIfNotNullOrDefault` a bit confused me, please change to this:
```
return "%s_%s".formatted(StringUtils.defaultString(eid.getInserter()), eid.getSource());
```